pythonbits/bb.py

```diff
@@ -76,28 +87,48 @@ def submit(payload):
         t = Tracker()
         return t.upload(**payload)
 
-    @form_field('scene', 'checkbox')
-    def _render_scene(self):
-        # todo: if path is directory, choose file for crc
-        path = os.path.normpath(self['path'])  # removes trailing slash
-        try:
-            try:
-                if os.path.exists(path) and not os.path.isdir(path):
-                    return is_scene_crc(path)
-            except KeyboardInterrupt:
-                sys.stdout.write('...skipped\n')
+    # TODO: These should be detected as scene releases and produce an error message
+    #       because they must be in a directory named after the release:
 
-            query_scene_fname(path)
-        except HTTPError as e:
-            log.notice(e)
+    # voa-the_omega_man_x264_bluray.mkv -> The.Omega.Man.1971.1080p.BluRay.x264-VOA/voa-the_omega_man_x264_bluray.mkv
+    # hd1080-wtl.mkv -> Walk.the.Line.Extended.Cut.2005.1080p.BluRay.x264-HD1080/hd1080-wtl.mkv
 
-        while True:
-            choice = input('Is this a scene release? [y/N] ')
-
-            if not choice or choice.lower() == 'n':
-                return False
-            elif choice.lower() == 'y':
-                return True
+    @form_field('scene', 'checkbox')
+    def _render_scene(self):
+        def ask_user():
+            return prompt_yesno('Is this a scene release?', default=False)
+
+        def handle_error(filepath, error):
+            log.error(str(error))
+
+        path = self['path']
+        modified = scene.check_integrity(path, on_error=handle_error)
```

Comment:
so if it thinks it's not a scene release, it queries the user every time?

The CRC check should stay as an option IMO, since it's the "safest". And it had the advantage of not querying the user if it wasn't found. My usecase for it is headless submissions.

Reply:
> so if it thinks it's not a scene release, it queries the user every time?

No. If it finds nothing, it's not a scene release. This should work all the time unless the release group or title were changed. This is an improvement over the old algorithm that always asked the user unless it found the exact title.

> The CRC check should stay as an option IMO

I don't object as long as I can disable it.